lib/fedora_migrate/repository_migrator.rb

```diff
@@ -15,34 +15,15 @@ def initialize namespace = nil, options = {}
 
     # TODO: need a reporting mechanism for results (issue #4)
     def migrate_objects
-      source_objects.each do |source|
-        Logger.info "Migrating source object #{source.pid}"
-        result = begin
-          FedoraMigrate::ObjectMover.new(source, nil, options).migrate
-        rescue NameError => e
-          "The most likely explanation is that your target is not defined"
-          error_message(e)
-        rescue StandardError => e
-          error_message(e)
-        end
-        unless result == true
-          Logger.warn "#{source.pid} failed.\n#{result}"
-          @failed = @failed + 1
-        end
-      end
+      source_objects.collect { |source| migrate_object(source) }
```

Comment:
Since you don't care about the array that this returns, I'd probably choose `#each` here over `#collect`. My understanding of `#collect`/`#map` is that they're most useful when you want to execute a proc over each item in an enumerable *and you care about the return values*.